bazel/google_cloud_cpp_deps.bzl

```diff
@@ -174,6 +181,7 @@ def google_cloud_cpp_deps(name = None):
         name = "com_github_nlohmann_json",
         urls = [
             "https://github.com/nlohmann/json/releases/download/v3.11.2/include.zip",
+            "https://storage.googleapis.com/cloud-cpp-community-archive/com_github_nlohmann_json/include.zip",
```

Comment:
Hmmm... this is unfortunate, without the version number in the name it will be unusable for older branches, or by any branch / PR that is out of date right after we move to v3.12.x.  Maybe rename this to `gs://.../v3.11.2.zip` and some ad-hoc code to handle this case?  I am not sure if this would work, but there are different tarball/zip files, e.g., https://github.com/nlohmann/json/archive/refs/tags/v3.11.2.tar.gz

Reply:
How about we simply download the full https://github.com/nlohmann/json/archive/v3.11.2.tar.gz archive, even though our `bazel/nlohmann_json.BUILD` file only uses the include files, and it is 7.8M instead of 288K? 
 Then there is no special case?

Reply:
The full download SGTM too.

Reply:
Done.